pdm/src/main/java/me/bristermitten/pdm/PDMBuilder.java

```diff
@@ -34,6 +38,38 @@ public PDMBuilder(@NotNull final Plugin plugin)
         applicationVersion(plugin.getDescription().getVersion());
     }
 
+    public PDMBuilder(@NotNull final Class<? extends Plugin> plugin)
+    {
+        Validate.isTrue("org.bukkit.plugin.java.PluginClassLoader".equals(plugin.getClassLoader().getClass().getName()),
+            "Plugin must be loaded with a PluginClassLoader");
+        classLoader((URLClassLoader) plugin.getClassLoader());
+        PluginDescriptionFile description = get(classLoader, "description");
+        dependenciesResource(classLoader.getResourceAsStream(DEPENDENCIES_RESOURCE_NAME));
+        rootDirectory(new File("./plugins"));
```

Comment:
Is there a cleaner way of referencing the directory here?

Reply:
We could get the plugin jar `File` from the class loader via reflection and use `getParentFile()`.

Reply:
I'm not sure if this is better or worse tbh. There's also the option of using `Bukkit.getUpdateFolderFile()` (and getting the parent file again), but I'm not 100% sure if that exists on older versions.

Reply:
It seems to exist on all versions.